### PR TITLE
add PPO example

### PR DESCRIPTION
**Pull request by zenghsh3**
- add PPO example
- refine DDPG code

### CODE REVIEW

File: parl/algorithms/ppo.py [hunk 1]
```diff
@@ -48,7 +48,8 @@ def _calc_logprob(self, actions, means, logvars):
         Returns:
             logprob: shape (batch_size)
         """
-        result1 = layers.elementwise_div(layers.square(actions - means), layers.exp(logvars), axis=1)
+        result1 = layers.elementwise_div(
```
Comment (TomorrowIsAnOtherDay) [inline parl/algorithms/ppo.py]: variable names such as `result1` and `result2` seem ambiguous here, could you rename them to another names that has substantial meaning actually.

File: parl/algorithms/ppo.py [hunk 2]
```diff
@@ -87,7 +86,7 @@ def define_predict(self, obs):
     def define_sample(self, obs):
         """ Use policy model of self.model to sample actions
         """
-        sampled_act = self.model.policy_sample(obs) 
+        sampled_act = self.model.policy_sample(obs)
         return sampled_act
 
     def define_policy_learn(self, obs, actions, advantages, beta=None):
```
Comment (TomorrowIsAnOtherDay) [inline parl/algorithms/ppo.py:93]: `Beta` decides which way to approximate TRPO, could you add some comments here?

File: examples/DDPG/train.py [hunk 3]
```diff
@@ -45,8 +45,7 @@ def run_train_episode(env, agent, rpm, act_bound):
         action = np.squeeze(action)
 
         # Add exploration noise, and clip to [-1.0, 1.0]
-        action = np.clip(
-            np.random.normal(action, 1.0), -1.0, 1.0)
+        action = np.clip(np.random.normal(action, 1.0), -1.0, 1.0)
         action = action * act_bound
```
Comment (TomorrowIsAnOtherDay) [inline examples/DDPG/train.py]: This line has a prerequisite that the action bound is symmetric, I recommend you to use a linear transformation here to transform the `[-1, -1]` to  `[lower bound, upper bound]`

File: parl/algorithms/ppo.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+from paddle import fluid
+from parl.framework.algorithm_base import Algorithm
+
+__all__ = ['PPO']
+
+
+class PPO(Algorithm):
+    def __init__(self, model, hyperparas):
+        Algorithm.__init__(self, model, hyperparas)
+        self.old_policy_model = deepcopy(model.policy_model)
```
Comment (TomorrowIsAnOtherDay) [inline parl/algorithms/ppo.py:28]: add some comments here, users may be curious at why there is copy of policy here

**Review by TomorrowIsAnOtherDay — CHANGES_REQUESTED** [review]

File: parl/algorithms/ppo.py (hunk 1 above)
Comment (zenghsh3) [reply parl/algorithms/ppo.py]: done

File: parl/algorithms/ppo.py (hunk 2 above)
Comment (zenghsh3) [reply parl/algorithms/ppo.py:93]: done

File: examples/DDPG/train.py (hunk 3 above)
Comment (zenghsh3) [reply examples/DDPG/train.py]: done

File: parl/algorithms/ppo.py (hunk 4 above)
Comment (zenghsh3) [reply parl/algorithms/ppo.py:28]: done

**Review by TomorrowIsAnOtherDay — APPROVED** [review]

**Pull request merged**